qiskit_nature/problems/second_quantization/lattice/lattice/triangular_lattice.py

```diff
@@ -0,0 +1,295 @@
+            # y direction
+            elif i == 1 and y != cols - 1:
+                node_b = _coordinate_to_index(np.array([x, y]) + np.array([0, 1]), size)
+            # diagonal direction
+            elif i == 2 and x != rows - 1 and y != cols - 1:
+                node_b = _coordinate_to_index(np.array([x, y]) + np.array([1, 1]), size)
+            else:
+                continue
+            list_of_edges.append((node_a, node_b, edge_parameter[i]))
+    return list_of_edges
+
+
+def _boundary_edges(
+    size: Tuple[int, int],
+    edge_parameter: Tuple[complex, complex, complex],
+    boundary_condition: BoundaryCondition,
+) -> List[Tuple[int, int, complex]]:
+    """Return a list consisting of the edges that cross the boundaries
+        depending on the boundary conditions.
+
+    Args:
+        size : Lengths of each dimension.
+        edge_parameter : Weights on the edges in each direction.
+        boundary_condition : Boundary condition for each dimension.
+            The available boundary conditions are:
+            BoundaryCondition.OPEN, BoundaryCondition.PERIODIC.
+    Raises:
+        ValueError: Given boundary condition is invalid values.
+    Returns:
+        List[Tuple[int, int, complex]]: List of weighted edges that cross the boundaries.
+    """
+    list_of_edges = []
+    rows, cols = size
+    # add edges when the boundary condition is periodic.
+    if boundary_condition == BoundaryCondition.PERIODIC:
+        # The periodic boundary condition in the x direction.
+        # It makes sense only when rows is greater than 2.
+        if rows > 2:
+            for y in range(cols):
+                node_a = (y + 1) * rows - 1
+                node_b = node_a - (rows - 1)  # node_b < node_a
+                list_of_edges.append((node_b, node_a, edge_parameter[0].conjugate()))
+        # The periodic boundary condition in the y direction.
+        # It makes sense only when cols is greater than 2.
+        if cols > 2:
+            for x in range(rows):
+                node_a = rows * (cols - 1) + x
+                node_b = x  # node_b < node_a
+                list_of_edges.append((node_b, node_a, edge_parameter[1].conjugate()))
+        # The periodic boundary condition in the diagonal direction.
+        for y in range(cols - 1):
+            node_a = (y + 1) * rows - 1
+            node_b = node_a + 1  # node_b > node_a
+            list_of_edges.append((node_a, node_b, edge_parameter[2]))
+
+        for x in range(rows - 1):
+            node_a = rows * (cols - 1) + x
+            node_b = x + 1  # node_b < node_a
+            list_of_edges.append((node_b, node_a, edge_parameter[2].conjugate()))
+
+        node_a = rows * cols - 1
+        node_b = 0  # node_b < node_a
+        list_of_edges.append((node_b, node_a, edge_parameter[2].conjugate()))
+
+    elif boundary_condition != BoundaryCondition.OPEN:
+        raise ValueError(
```

Comment:
Same suggestion as before:
```suggestion
    elif boundary_condition == BoundaryCondition.OPEN:
        continue
    else:
        raise ValueError(
```

Reply:
Thanks for raising the point. I have rewritten it, but in this case I couldn't use `continue`, so instead I have used `pass` here.